tests/test_types_typeddict.py

```diff
@@ -711,6 +711,63 @@ class RecursiveGenTypedDict(TypedDict, Generic[T]):
     ]
 
 
+def test_nested_generic_typeddict():
+    T = TypeVar('T')
+    U = TypeVar('U')
+
+    class BaseGenericTypedDict(TypedDict, Generic[U]):
+        type: U
+
+    class NestedGenericTypedDict(BaseGenericTypedDict[T]):
+        value: T
+
+    TypeAdapter(NestedGenericTypedDict[str]).validate_python(
+        {
+            'type': 'string',
+            'value': 'string',
+        }
+    )
```

Comment:
Can you elaborate on why this test? It seems like a simplified case of the below, and it doesn't seem to fail on `main` either.

Reply:
I expanded on it here https://github.com/pydantic/pydantic/issues/7988#issuecomment-1795561417 , I should probably confirm the schema is correct and/or test with TypeVar defaults

Reply:
That comment definitely applies for the test below, but I didn't follow what the problem was in this reduced case.

Reply:
Oh in that case I can properly check this one when I got home

Reply:
This test was meant to ensure that `type: U` (renamed the key to `format`) was correctly being set type `str`. Currently it sets it to `any` which is why it passes.